bokehjs/src/lib/models/canvas/canvas.ts

```diff
@@ -114,13 +114,13 @@ export class CanvasView extends DOMView {
   }
 
   prepare_canvas(width: number, height: number): void {
-    // Ensure canvas has the correct size, taking HIDPI into account
     this.bbox = new BBox({left: 0, top: 0, width, height})
 
     this.el.style.width = `${width}px`
     this.el.style.height = `${height}px`
 
-    const pixel_ratio = get_scale_ratio(this.ctx, this.model.use_hidpi, this.model.output_backend)
+    const {use_hidpi, output_backend} = this.model
+    const pixel_ratio = use_hidpi && output_backend != "svg" ? devicePixelRatio : 1
```

Comment:
Where is the value of `devicePixelRatio` coming from?

Reply:
`window`, or the global scope in general. Given that we don't rely on jsdom and nodejs for testing anymore, we can refer to all globals equally.

Reply:
What specifically has changed that we can drop the old method of computing the pixel ratio?

Reply:
`backingStorePixelRatio` was never part of canvas specification, so it shouldn't have been used in the first place. Its usage is deprecated. As far as I can tell, if it was available at all (as a prefixed property), its value was usually 1, thus not contributing anything. Perhaps some older variants of some browsers (Safari?) messed with this setting to some extent.